Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/CachedQueries/RegistryMapFallback.hs

```diff
@@ -0,0 +1,69 @@
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Storage.CachedQueries.RegistryMapFallback where
+
+import Domain.Types.Merchant (Merchant)
+import Domain.Types.RegistryMapFallback
+import Kernel.Prelude
+import qualified Kernel.Storage.Hedis as Hedis
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import Storage.Queries.RegistryMapFallback as Queries
+
+findBySubscriberId :: (CacheFlow m r, MonadFlow m) => Text -> m [RegistryMapFallback]
+findBySubscriberId subscriberId =
+  Hedis.safeGet (makeSubscriberIdKey subscriberId) >>= \case
+    Just a -> return a
+    Nothing -> cacheRegistryMapFallbacks (makeSubscriberIdKey subscriberId) /=<< Queries.findBySubscriberId subscriberId
+
+findByUniqueId :: (CacheFlow m r, MonadFlow m) => Text -> m [RegistryMapFallback]
+findByUniqueId uniqueId =
+  Hedis.safeGet (makeUniqueIdKey uniqueId) >>= \case
+    Just a -> return a
+    Nothing -> cacheRegistryMapFallbacks (makeUniqueIdKey uniqueId) /=<< Queries.findByUniqueId uniqueId
+
+findByMerchantId :: (CacheFlow m r, MonadFlow m) => Id Merchant -> m (Maybe RegistryMapFallback)
+findByMerchantId merchantId =
+  Hedis.safeGet (makeMerchantIdKey merchantId) >>= \case
+    Just a -> return a
+    Nothing -> cacheRegistryMapFallback (makeMerchantIdKey merchantId) /=<< Queries.findByMerchantId merchantId
+
+findBySubscriberIdAndUniqueId :: (CacheFlow m r, MonadFlow m) => Text -> Text -> m (Maybe RegistryMapFallback)
+findBySubscriberIdAndUniqueId subId uniqueId =
+  Hedis.safeGet (makeSubscriberIdAndUniqueIdKey subId uniqueId) >>= \case
+    Just a -> return a
+    Nothing -> cacheRegistryMapFallback (makeSubscriberIdAndUniqueIdKey subId uniqueId) /=<< Queries.findBySubscriberIdAndUniqueId subId uniqueId
+
+cacheRegistryMapFallbacks :: CacheFlow m r => Text -> [RegistryMapFallback] -> m ()
+cacheRegistryMapFallbacks key registryMap = do
+  expTime <- fromIntegral <$> asks (.cacheConfig.configsExpTime)
+  Hedis.setExp key registryMap expTime
+
+cacheRegistryMapFallback :: CacheFlow m r => Text -> Maybe RegistryMapFallback -> m ()
+cacheRegistryMapFallback key registryMap = do
+  expTime <- fromIntegral <$> asks (.cacheConfig.configsExpTime)
+  Hedis.setExp key registryMap expTime
+
+makeSubscriberIdKey :: Text -> Text
+makeSubscriberIdKey subscriberId = "driver-offer:CachedQueries:RegistryMapFallback:SubscriberId-" <> subscriberId
+
+makeUniqueIdKey :: Text -> Text
+makeUniqueIdKey uniqueId = "driver-offer:CachedQueries:RegistryMapFallback:uniqueId-" <> uniqueId
```

Comment:
Make U capital, to follow the semantics.

Reply:
done